DL4J uses SharedTrainingMaster on spark and reports "ERROR - publicationUnblockTimeoutNs=15000000000 <= clientLivenessTimeoutNs=30000000000"

#### Issue Description

The node IP is 10.0.6.201~204. When configuring VoidConfiguration, set networkMask=10.0.6.0/16, and the UDP port is 40123. When deployed in spark Standalone mode, it keeps reporting an error: "Caused by: io.aeron.exceptions.ConfigurationException: ERROR - publicationUnblockTimeoutNs=15000000000 <= clientLivenessTimeoutNs=30000000000", the spark cluster server firewall is all closed.

中文描述：
节点IP是10.0.6.201~204，配置VoidConfiguration时，设置networkMask=10.0.6.0/16，UDP端口为40123，spark Standalone 模式部署下时，一直报错：“Caused by: io.aeron.exceptions.ConfigurationException: ERROR - publicationUnblockTimeoutNs=15000000000 <= clientLivenessTimeoutNs=30000000000”，spark集群服务器防火墙是全部关闭的。

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version：1.0.0-M2
* platform information (OS, etc)：Centos7.9
* CUDA version, if used：Use CPUs
* NVIDIA driver version, if in use

#### Contributing

If you'd like to help us fix the issue by contributing some code, but would
like guidance or help in doing so, please mention it!

#### Issue Description

The node IP is 10.0.6.201~204. When configuring VoidConfiguration, set networkMask=10.0.6.0/16, and the UDP port is 40123. When deployed in spark Standalone mode, it keeps reporting an error: "Caused by: io.aeron.exceptions.ConfigurationException: ERROR - publicationUnblockTimeoutNs=15000000000 <= clientLivenessTimeoutNs=30000000000", the spark cluster server firewall is all closed.

中文描述：
节点IP是10.0.6.201~204，配置VoidConfiguration时，设置networkMask=10.0.6.0/16，UDP端口为40123，spark Standalone 模式部署下时，一直报错：“Caused by: io.aeron.exceptions.ConfigurationException: ERROR - publicationUnblockTimeoutNs=15000000000 <= clientLivenessTimeoutNs=30000000000”，spark集群服务器防火墙是全部关闭的。

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version：1.0.0-M2
* platform information (OS, etc)：Centos7.9
* CUDA version, if used：Use CPUs
* NVIDIA driver version, if in use

#### Contributing

If you'd like to help us fix the issue by contributing some code, but would
like guidance or help in doing so, please mention it!

--- a/deeplearning4j/deeplearning4j-scaleout/deeplearning4j-scaleout-parallelwrapper-parameter-server/src/main/java/org/deeplearning4j/parallelism/parameterserver/ParameterServerTrainerContext.java
+++ b/deeplearning4j/deeplearning4j-scaleout/deeplearning4j-scaleout-parallelwrapper-parameter-server/src/main/java/org/deeplearning4j/parallelism/parameterserver/ParameterServerTrainerContext.java
@@ -68,7 +68,7 @@ public class ParameterServerTrainerContext implements TrainerContext {
      * @param rootDevice the root device id
      * @param useMDS     whether to use MultiDataSet or DataSet
      *                   or not
-     * @param wrapper    the wrapper instance to use with this trainer (this refernece is needed
+     * @param wrapper    the wrapper instance to use with this trainer (this reference is needed
      *                   for coordination with the {@link ParallelWrapper} 's {@link TrainingListener}
      * @return the created training instance
      */

--- a/nd4j/nd4j-parameter-server-parent/nd4j-parameter-server-client/pom.xml
+++ b/nd4j/nd4j-parameter-server-parent/nd4j-parameter-server-client/pom.xml
@@ -72,7 +72,7 @@
         <dependency>
             <groupId>org.zeroturnaround</groupId>
             <artifactId>zt-exec</artifactId>
-            <version>1.9</version>
+            <version>1.12</version>
             <scope>test</scope>
         </dependency>
         <dependency>

--- a/nd4j/nd4j-parameter-server-parent/nd4j-parameter-server-node/src/main/java/org/nd4j/parameterserver/distributed/v2/transport/impl/AeronUdpTransport.java
+++ b/nd4j/nd4j-parameter-server-parent/nd4j-parameter-server-node/src/main/java/org/nd4j/parameterserver/distributed/v2/transport/impl/AeronUdpTransport.java
@@ -20,6 +20,7 @@
 
 package org.nd4j.parameterserver.distributed.v2.transport.impl;
 
+import io.aeron.driver.Configuration;
 import org.nd4j.shade.guava.math.IntMath;
 import io.aeron.Aeron;
 import io.aeron.FragmentAssembler;
@@ -118,13 +119,17 @@ public class AeronUdpTransport extends BaseTransport implements AutoCloseable {
 
         Preconditions.checkArgument(ownPort > 0 && ownPort < 65536, "Own UDP port should be positive value in range of 1 and 65536");
         Preconditions.checkArgument(rootPort > 0 && rootPort < 65536, "Master node UDP port should be positive value in range of 1 and 65536");
+        if(!System.getProperties().containsKey(Configuration.CLIENT_LIVENESS_TIMEOUT_PROP_NAME)) {
+            //see: https://github.com/eclipse/deeplearning4j/issues/9677, ensure that publication unblock is >= client liveness
+            setProperty(Configuration.CLIENT_LIVENESS_TIMEOUT_PROP_NAME, "30000000000");
+            setProperty(Configuration.PUBLICATION_UNBLOCK_TIMEOUT_PROP_NAME,"35000000000");
 
-        setProperty("aeron.client.liveness.timeout", "30000000000");
+        }
 
         // setting this property to try to increase maxmessage length, not sure if it still works though
         //Term buffer length: must be power of 2 and in range 64kB to 1GB: https://github.com/real-logic/aeron/wiki/Configuration-Options
         String p = System.getProperty(ND4JSystemProperties.AERON_TERM_BUFFER_PROP);
-        if(p == null){
+        if(p == null) {
             System.setProperty(ND4JSystemProperties.AERON_TERM_BUFFER_PROP, String.valueOf(DEFAULT_TERM_BUFFER_PROP));
         }
 
@@ -449,40 +454,40 @@ public class AeronUdpTransport extends BaseTransport implements AutoCloseable {
             // if response != OK we must do something with response
             switch (status) {
                 case MAX_POSITION_EXCEEDED: {
-                        log.warn("MaxPosition hit: [{}]", id);
-                        try {
-                            // in case of backpressure we're just sleeping for a while, and message out again
-                            Thread.sleep(voidConfiguration.getRetransmitTimeout());
-                        } catch (InterruptedException e) {
-                            //
-                        }
+                    log.warn("MaxPosition hit: [{}]", id);
+                    try {
+                        // in case of backpressure we're just sleeping for a while, and message out again
+                        Thread.sleep(voidConfiguration.getRetransmitTimeout());
+                    } catch (InterruptedException e) {
+                        //
                     }
-                    break;
+                }
+                break;
                 case CLOSED: {
                     // TODO: here we should properly handle reconnection
                     log.warn(" Connection was closed: [{}]", id);
                     return;
                 }
                 case ADMIN_ACTION: {
-                        log.info("ADMIN_ACTION: [{}]", id);
-                        try {
-                            Thread.sleep(voidConfiguration.getRetransmitTimeout());
-                        } catch (InterruptedException e) {
-                            //
-                        }
+                    log.info("ADMIN_ACTION: [{}]", id);
+                    try {
+                        Thread.sleep(voidConfiguration.getRetransmitTimeout());
+                    } catch (InterruptedException e) {
+                        //
                     }
-                    break;
+                }
+                break;
                 case NOT_CONNECTED: {
-                            log.info("NOT_CONNECTED: [{}]", id);
-                            addConnection(id);
-                            try {
-                                // in case of backpressure we're just sleeping for a while, and message out again
-                                Thread.sleep(voidConfiguration.getRetransmitTimeout());
-                            } catch (InterruptedException e) {
-                                //
-                            }
-                        }
-                        break;
+                    log.info("NOT_CONNECTED: [{}]", id);
+                    addConnection(id);
+                    try {
+                        // in case of backpressure we're just sleeping for a while, and message out again
+                        Thread.sleep(voidConfiguration.getRetransmitTimeout());
+                    } catch (InterruptedException e) {
+                        //
+                    }
+                }
+                break;
                 case BACK_PRESSURED: {
                     log.info("BACK_PRESSURED: [{}]", id);
                     try {

--- a/nd4j/nd4j-parameter-server-parent/nd4j-parameter-server-node/src/main/resources/aeron.properties
+++ b/nd4j/nd4j-parameter-server-parent/nd4j-parameter-server-node/src/main/resources/aeron.properties
@@ -20,4 +20,5 @@
 
 aeron.mtu.length=8192
 aeron.client.liveness.timeout=30000000000
+aeron.publication.unblock.timeout=35000000000
 aeron.image.liveness.timeout=30000000000

--- a/nd4j/nd4j-parameter-server-parent/nd4j-parameter-server-node/src/test/resources/aeron.properties
+++ b/nd4j/nd4j-parameter-server-parent/nd4j-parameter-server-node/src/test/resources/aeron.properties
@@ -19,9 +19,10 @@
 #
 
 aeron.mtu.length=16384,
-aeron.socket.so_sndbuf=2097152,
-aeron.socket.so_rcvbuf=2097152,
-aeron.rcv.buffer.length=16384,
+aeron.socket.so_sndbuf=2097152
+aeron.socket.so_rcvbuf=2097152
+aeron.rcv.buffer.length=16384
 aeron.rcv.initial.window.length=2097152
 aeron.client.liveness.timeout=30000000000
+aeron.publication.unblock.timeout=35000000000
 aeron.image.liveness.timeout=30000000000

--- a/nd4j/nd4j-serde/nd4j-aeron/pom.xml
+++ b/nd4j/nd4j-serde/nd4j-aeron/pom.xml
@@ -39,7 +39,7 @@
 
 
         <sbe.version>1.5.4</sbe.version>
-        <aeron.version>1.32.0</aeron.version>
+        <aeron.version>1.38.1</aeron.version>
         <module.name>nd4j.aeron</module.name>
     </properties>
 

--- a/nd4j/nd4j-serde/nd4j-aeron/src/main/java/org/nd4j/aeron/ipc/AeronUtil.java
+++ b/nd4j/nd4j-serde/nd4j-aeron/src/main/java/org/nd4j/aeron/ipc/AeronUtil.java
@@ -38,13 +38,24 @@ import java.util.function.Consumer;
 
 public class AeronUtil {
 
+
+    /**
+     * Sets a system property if not set
+     * @param propertyName the name of the property
+     * @param value the value to set
+     */
+    public static void setSystemPropertyIfNotSet(String propertyName,String value) {
+        if(!System.getProperties().containsKey(propertyName)) {
+            System.setProperty(propertyName,value);
+        }
+    }
     /**
      * Get a media driver context
      * for sending ndarrays
      * based on a given length
      * where length is the length (number of elements)
-     * in the ndarrays hat are being sent
-     * @param length the length to based the ipc length
+     * in the ndarrays that are being sent
+     * @param length the length to base the ipc length
      * @return the media driver context based on the given length
      */
     public static MediaDriver.Context getMediaDriverContext(int length) {
@@ -57,14 +68,10 @@ public class AeronUtil {
         //ipc length must be positive power of 2
         while (!BitUtil.isPowerOfTwo(ipcLength))
             ipcLength += 2;
-        // System.setProperty("aeron.term.buffer.size",String.valueOf(ipcLength));
         final MediaDriver.Context ctx =
                 new MediaDriver.Context().threadingMode(ThreadingMode.SHARED)
                         .dirDeleteOnStart(true)
                         .dirDeleteOnShutdown(true)
-                        /*  .ipcTermBufferLength(ipcLength)
-                          .publicationTermBufferLength(ipcLength)
-                          .maxTermBufferLength(ipcLength)*/
                         .conductorIdleStrategy(new BusySpinIdleStrategy())
                         .receiverIdleStrategy(new BusySpinIdleStrategy())
                         .senderIdleStrategy(new BusySpinIdleStrategy());

--- a/nd4j/nd4j-serde/nd4j-aeron/src/main/java/org/nd4j/aeron/ipc/LowLatencyMediaDriver.java
+++ b/nd4j/nd4j-serde/nd4j-aeron/src/main/java/org/nd4j/aeron/ipc/LowLatencyMediaDriver.java
@@ -27,6 +27,7 @@ import org.agrona.concurrent.SigIntBarrier;
 
 import static java.lang.System.setProperty;
 import static org.agrona.concurrent.UnsafeBuffer.DISABLE_BOUNDS_CHECKS_PROP_NAME;
+import static org.nd4j.aeron.ipc.AeronUtil.setSystemPropertyIfNotSet;
 
 public class LowLatencyMediaDriver {
 
@@ -35,11 +36,11 @@ public class LowLatencyMediaDriver {
     @SuppressWarnings("checkstyle:UncommentedMain")
     public static void main(final String... args) {
         MediaDriver.main(args);
-        setProperty(DISABLE_BOUNDS_CHECKS_PROP_NAME, "true");
-        setProperty("aeron.mtu.length", "16384");
-        setProperty("aeron.socket.so_sndbuf", "2097152");
-        setProperty("aeron.socket.so_rcvbuf", "2097152");
-        setProperty("aeron.rcv.initial.window.length", "2097152");
+        setSystemPropertyIfNotSet(DISABLE_BOUNDS_CHECKS_PROP_NAME, "true");
+        setSystemPropertyIfNotSet("aeron.mtu.length", "16384");
+        setSystemPropertyIfNotSet("aeron.socket.so_sndbuf", "2097152");
+        setSystemPropertyIfNotSet("aeron.socket.so_rcvbuf", "2097152");
+        setSystemPropertyIfNotSet("aeron.rcv.initial.window.length", "2097152");
 
         final MediaDriver.Context ctx =
                 new MediaDriver.Context().threadingMode(ThreadingMode.DEDICATED).dirDeleteOnStart(true)

